File: rebus-rebuscrus/src/java/it/csi/rebus/rebuscrus/vo/AllegatoVeicoloVO.java
/*
* SPDX-FileCopyrightText: (C) Copyright 2022 Regione Piemonte
* SPDX-License-Identifier: EUPL-1.2
*/
package it.csi.rebus.rebuscrus.vo;

import java.util.Date;

@SuppressWarnings("serial")
public class AllegatoVeicoloVO extends ParentVO {
	private TipoDocumentoVO tipoDocumento;
	private String note;
	private byte[] file;
	private String nomeFile;
	private Date dataCaricamento;

	public TipoDocumentoVO getTipoDocumento() {
		return tipoDocumento;
	}

	public void setTipoDocumento(TipoDocumentoVO tipoDocumento) {
		this.tipoDocumento = tipoDocumento;
	}

	public String getNote() {
		return note;
	}

	public void setNote(String note) {
		this.note = note;
	}

	public byte[] getFile() {
		return file;
	}

	public void setFile(byte[] file) {
		this.file = file;
	}

	public String getNomeFile() {
		return nomeFile;
	}

	public void setNomeFile(String nomeFile) {
		this.nomeFile = nomeFile;
	}

	public Date getDataCaricamento() {
		return dataCaricamento;
	}

	public void setDataCaricamento(Date dataCaricamento) {
		this.dataCaricamento = dataCaricamento;
	}
}

File: rebus-rebuscruswcl/src/app/vo/allegato-veicolo-vo.ts
/*************************************************
Copyright Regione Piemonte - 2022
SPDX-License-Identifier: EUPL-1.2-or-later
***************************************************/
import { TipoDocumentoVO } from "./extend-vo";

export class AllegatoVeicoloVO {
    tipoDocumento: TipoDocumentoVO;
    note: string;
    file: File;
    nomeFile: string;
    dataCaricamento: Date;
    primoTelaioVeicolo: string;
    nTargaVeicolo: string;

    constructor(){
        if(!this.tipoDocumento) this.tipoDocumento = new TipoDocumentoVO(); 
    }
}